packages/builder/src/components/automation/AutomationBuilder/FlowChart/FlowItemHeader.svelte

```diff
@@ -1,27 +1,39 @@
 <script>
-  import { automationStore } from "builderStore"
-  import { Icon, Body, Detail, StatusLight } from "@budibase/bbui"
+  import { automationStore, selectedAutomation } from "builderStore"
+  import { Icon, Body, StatusLight, AbsTooltip } from "@budibase/bbui"
   import { externalActions } from "./ExternalActions"
   import { createEventDispatcher } from "svelte"
+  import { Features } from "constants/backend/automations"
 
   export let block
   export let open
   export let showTestStatus = false
   export let testResult
   export let isTrigger
   export let idx
+  export let addLooping
+  export let deleteStep
+
+  let validRegex = /^[A-Za-z0-9_\s]+$/
+  let typing = false
 
   const dispatch = createEventDispatcher()
 
+  $: stepNames = $selectedAutomation.definition.stepNames
+  $: automationName = stepNames[block.id] || block.name || ""
+  $: automationNameError = getAutomationNameError(automationName)
+  $: status = updateStatus(testResult, isTrigger)
+  $: isTrigger = isTrigger || block.type === "TRIGGER"
```

Comment:
This is a recursive declaration, so I'm not sure how svelte will handle this. I think we'd be better using a new variable name here to distinguish between the prop and this, and it would also help keep things more understandable.

I can see that you just moved this, but still think it's worth addressing since we're using it in other ways now.